Documentation/quickstart/debugging.rst

```diff
@@ -6,35 +6,40 @@ Debugging
 
 Finding and fixing bugs is an important part of the hardware and software development process. Sometimes you also need
 to use debugging techniques to understand how the system works. Two tools that are helpful are debug logging and
-debugging using the Gnu Debugger (gdb).
+debugging using the GNU Debugger (gdb).
 
 Debug Logging
 -------------
 
 NuttX has a powerful logging facility with ``info``, ``warn``, and ``error`` levels. You can enable debugging for your
-build for the ``net`` feature (TCP/IP stack) by putting the following lines in your ``.config`` file:
-
-    ::
-
-       CONFIG_DEBUG_ALERT=y
-       CONFIG_DEBUG_FEATURES=y
-       CONFIG_DEBUG_ERROR=y
-       CONFIG_DEBUG_WARN=y
-       CONFIG_DEBUG_INFO=y
-       CONFIG_DEBUG_ASSERTIONS=y
-       CONFIG_DEBUG_NET=y
-       CONFIG_DEBUG_NET_ERROR=y
-       CONFIG_DEBUG_NET_WARN=y
-       CONFIG_DEBUG_NET_INFO=y
-       CONFIG_DEBUG_SYMBOLS=y
-       CONFIG_DEBUG_NOOPT=y
-       CONFIG_SYSLOG_TIMESTAMP=y
-
-Note that turning all these to ``y`` will produce an incredible amount of logging output. Set the level you want and
-the area you're interested in to ``y``, and the rest to ``n``, and then recompile. You can see the full list of
+build for the ``net`` feature (TCP/IP stack) by running this simple script:
+
+.. code-block:: console
+
+   #!/bin/bash
+
+   kconfig-tweak --enable CONFIG_DEBUG_ALERT
+   kconfig-tweak --enable CONFIG_DEBUG_FEATURES
+   kconfig-tweak --enable CONFIG_DEBUG_ERROR
+   kconfig-tweak --enable CONFIG_DEBUG_WARN
+   kconfig-tweak --enable CONFIG_DEBUG_INFO
+   kconfig-tweak --enable CONFIG_DEBUG_ASSERTIONS
+   kconfig-tweak --enable CONFIG_DEBUG_NET
+   kconfig-tweak --enable CONFIG_DEBUG_NET_ERROR
+   kconfig-tweak --enable CONFIG_DEBUG_NET_WARN
+   kconfig-tweak --enable CONFIG_DEBUG_NET_INFO
+   kconfig-tweak --enable CONFIG_DEBUG_SYMBOLS
+   kconfig-tweak --enable CONFIG_DEBUG_NOOPT
+
+Note that enabling all these will produce an incredible amount of logging output. Enable the level you want and
```

Comment:
Is there a reason we are not recommending that people explore the menuconfig?  I get that `kconfig-tweak` is useful for turning things on and off, but searching through header files seems to be working against the while point of menuconfig.

Reply:
@btashton Well, when I wrote this, I found using the menuconfig system to turn on and off logging levels to be incredibly frustrating. I found navigating and finding things very difficult. So I resorted to this system. 

If you have another way to easily turn on and off logging, or we can refactor the menuconfig system to make this easier, I'm in. Let me know.

Reply:
I just turn on all the logging levels and then enable logging level for whatever system I need.  The only gotcha here is that you need to turn on the global levels first.  I also keep that enabled in my defconfig along with things like debug symbols until I am ready to upstream.

My biggest objection is that using things like kconfig-tweak and searching header files is that people do not resolve dependencies correctly or find useful features for what they have enabled.  Plus you have docs in menuconfig.

Reply:
I also would like to add some https://asciinema.org/ demos which we can embed in the docs, but that is for later.
https://pypi.org/project/sphinxcontrib.asciinema/

Reply:
@btashton If you have a vision for how to rewrite this section, I would love to see that! I would love to learn a better way. I am a NuttX newbie. 

I wrote this after being frustrated from a series of long hard driver debugging sessions, from the perspective of someone who knew nothing about NuttX internals and had no guidance about how to learn what was happening inside NuttX. This was my best attempt at telling others information that I wanted to have...

Reply:
@btashton The asciinema stuff sounds great too.

Reply:
@adamfeuer Sounds good, we can leave this as is lets just make sure we address the custinfo thing.